Wkhtmltopdf.cfc

```diff
@@ -123,7 +123,9 @@ component {
   private function _quote(required string val) {
     // escape and quote the value if it is a string and this isn't windows
     if (server.os.name != 'UNIX') {
-      val = reReplace(val, '(["\\$`])', '\\\1', 'all');
+      if(val CONTAINS " "){
```

Comment:
Hi Loïc,

I am using this on Windows with wkhtmltopdf 0.12.2.1 (with patched qt) and
using Lucee 4.5.3.018 and I have the following code:

<cfset local.pdfArgs = {
'marginTop': 20,
'headerSpacing': 10,
'headerFontSize': 8,
'headerLeft': "Strategic Housing for 'Older People' Analysis Tool - SHOP@",
'headerRight': '#local.authorityName# - #dateFormat(now(), "dd/mm/yy")#',
'footerHtml': #application.config.getPath("ROOTURL")# &
"system/shopTool/pdf_footer.html",
'footerSpacing': 10,
'marginBottom': 35,
'marginLeft': 20,
'marginRight': 20
}>
<cfset local.pdf =
local.wkhtmltopdf.fromString(request.page.render("SHOPToolPrint"),
local.pdfArgs)>


With the regular expression in place I get an error associated with the
headerLeft argument: The pipe has been ended

With the regular expression in place and
if(val CONTAINS " "){
  return '"#val#"';
}

... the following shows up in the header:
Strategic Housing for \\'Older People\\' Analysis Tool - SHOP@

The double \\ is as a result of a difference in the way the regex is parsed
between Lucee and ACF. Even so - correcting this output still was not right.

Then removing the regular expression results in:
Strategic Housing for 'Older People' Analysis Tool - SHOP@


However, this can still break if a double quote is used in the string...

I am now using:
val = replace(val, '"', '\"', 'all');
if(val CONTAINS " "){
   return '"#val#"';
}

This has worked with everything that I have thrown at it - including the
characters that were being escaped in the regex.


All the best,

Martin


On 9 August 2016 at 11:13, Loïc Mahieu <notifications@github.com> wrote:

> In Wkhtmltopdf.cfc
> <https://github.com/LoicMahieu/cf-wkhtmltopdf/pull/1#discussion_r74032231>
> :
>
> > @@ -123,7 +123,9 @@ component {
> >    private function _quote(required string val) {
> >      // escape and quote the value if it is a string and this isn't windows
> >      if (server.os.name != 'UNIX') {
> > -      val = reReplace(val, '(["\\$`])', '\\\1', 'all');
> > +      if(val CONTAINS " "){
>
> Is there any reason about this change ? We had to escape potential quotes.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/LoicMahieu/cf-wkhtmltopdf/pull/1/files/3611e9bf5ffd6a1943b96a4ffb3f22fc4451ab85#r74032231>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFBJ-wxVyuhVUfpMdEdIvJ-oG5ry4GbIks5qeFLagaJpZM4JbyvM>
> .
>



-- 
Martin Webb
Director, CUBICstate Ltd.

Tel: +44 (0)1422 352233
Web: www.cubicstate.com
--